src/core/data/worker.c

```diff
@@ -153,6 +175,17 @@ _worker_pipe_write(void)
         log_error("could not write to pipe - %s", strerror(pipe_term->err));
     }
 }
+#endif
+
+static inline void
+_worker_notify_server(void)
+{
+#ifdef USE_EVENT_FD
+    _worker_write_event_fd();
```

Comment:
And similar to comment to the server side code, maybe just expand everything in this one function.